README.md

```diff
@@ -0,0 +1,1743 @@
+
+<h4 id="server_provided_all_addons_hash">server_provided_all_addons_hash</h4>
+
+
+The aggregate hash of all addons that this Agent previously received from the server via AddonsAvailable message.
+
+The server SHOULD compare this hash to the aggregate hash of all addons that it has for this Agent and if the hashes are different the server SHOULD send an AddonsAvailable message to the agent.
+
+<h3 id="agentaddonstatus-message">AgentAddonStatus Message</h3>
+
+
+The AgentAddonStatus has the following structure:
+
+
+```protobuf
+message AgentAddonStatus {
+    string name = 1;
+    bytes agent_has_hash = 2;
+    bytes server_offered_hash = 3;
+    enum Status {
+        INSTALLED = 0;
+        INSTALLING = 1;
+        INSTALL_FAILED = 2;
+    }
+    Status status = 4;
+    string error_message = 5;
+}
+```
+
+
+<h4 id="name">name</h4>
+
+
+Addon name. MUST be always set and MUST match the key in the addons field of AgentAddonStatuses message.
+
+<h4 id="agent_has_hash">agent_has_hash</h4>
+
+
+The hash of the addon that the Agent has.
+
+MUST be set if the Agent has this addon.
+
+MUST be empty if the Agent does not have this addon. This may be the case for example if the addon was offered by the Server but failed to install and the agent did not have this addon previously.
+
+<h4 id="server_offered_hash">server_offered_hash</h4>
+
+
+The hash of the addon that the server offered to the agent.
+
+MUST be set if the installation of the addon is initiated by an earlier offer from the server to install this addon. 
+
+MUST be empty if the Agent has this addon but it was installed locally and was not offered by the server.
+
+Note that it is possible for both agent_has_hash and server_offered_hash fields to be set and to have different values. This is for example possible if the agent already has a version of the addon successfully installed, the server offers a different version, but the agent fails to install that version.
+
+<h4 id="status">status</h4>
+
+
+The status of this addon. The possible values are:
+
+INSTALLED: Addon is successfully installed by the Agent. The error_message field MUST NOT be set.
+
+INSTALLING: Agent is currently downloading and installing the addon. server_offered_hash field MUST be set to indicate the version that the agent is installing. The error_message field MUST NOT be set.
+
+INSTALL_FAILED: Agent tried to install the addon but installation failed. server_offered_hash field MUST be set to indicate the version that the agent tried to install. The error_message may also contain more details about the failure.
+
+<h4 id="error_message">error_message</h4>
+
+
+An error message if the status is erroneous.
+
+<h3 id="agentinstallstatus-message">AgentInstallStatus Message</h3>
+
+
+This message contains the status of the last agent package install status performed by the agent and has the following structure:
+
+
+```protobuf
+message AgentInstallStatus {
+    bytes server_offered_hash = 1;
```

Comment:
I think we are missing here the additional version information which we discussed in the Google Doc :) 
```suggestion
    bytes server_offered_version = 1;
    bytes server_offered_hash = 2;
```
(and so on)

Reply:
Yes, I will add changes as separate PRs.